pkg/apiserver/webhooks/datavolume-mutate.go

```diff
@@ -71,7 +74,20 @@ func (wh *dataVolumeMutatingWebhook) Admit(ar admissionv1.AdmissionReview) *admi
 		return toAdmissionResponseError(err)
 	}
 
-	pvcSource := dataVolume.Spec.Source.PVC
+	if dataVolume.Spec.Source != nil {
+		pvcSource = dataVolume.Spec.Source.PVC
+	} else if dataVolume.Spec.SourceRef != nil && dataVolume.Spec.SourceRef.Kind == cdiv1.DataVolumeDataSource {
+		ns := dataVolume.Namespace
+		if dataVolume.Spec.SourceRef.Namespace != nil && *dataVolume.Spec.SourceRef.Namespace != "" {
+			ns = *dataVolume.Spec.SourceRef.Namespace
+		}
+		dataSource, err := wh.cdiClient.CdiV1beta1().DataSources(ns).Get(context.TODO(), dataVolume.Spec.SourceRef.Name, metav1.GetOptions{})
```

Comment:
I'll sure add it, although not sure it's part of the req/design.

Reply:
Won't add it, follwing our discussion with @awels .

Reply:
Normally I would agree with @mhenriks but we are already checingk if source PVCs exists for clone, and since clone sources are going to be the majority of datasource, we should remain consistent.

Reply:
Is this Get using a watch cache?
Depending on how often we think we'll see these cases it might be better to not cause api requests for this if we can have a watcher instead. Not sure how CDI does that right now.

Reply:
Not cached but I don't think we expect high volume.  At least initially